### PR TITLE
[XLA:GPU] remove unused buffers to fix cudnn graph

### PR DESCRIPTION
**Pull request by Cjkkkk**
This PR is to fix the bug introduced in commit https://github.com/openxla/xla/pull/5184/commits/25aa1baa87abb259e3e0350737c4d39a715183a0 in runner clean up PR: https://github.com/openxla/xla/pull/5184.

Unused buffers in fused attention need to be removed from data_vec so cudnn graph finalization doesn't error out.

### CODE REVIEW

File: xla/stream_executor/cuda/cuda_dnn.cc [hunk 1]
```diff
@@ -6734,6 +6734,14 @@ class CudnnExecutionPlanRunner<void(Args...)>
       data_ptrs_vec.pop_back();
     }
 
+    if (sizeof...(Args) == 7 || sizeof...(Args) == 11) {
+      // is fused attention fwd and bwd
+      // remove empty buffers from the list
+      data_ptrs_vec.erase(std::remove(data_ptrs_vec.begin(), data_ptrs_vec.end(), nullptr), data_ptrs_vec.end());
```
Comment (ezhulenev) [inline xla/stream_executor/cuda/cuda_dnn.cc:6740]: This needs formatting to be <= 80 characters

Comment (Cjkkkk) [reply xla/stream_executor/cuda/cuda_dnn.cc]: fixed

**Review by ezhulenev — APPROVED** [review]

**Comment by Cjkkkk** [comment]
@ezhulenev could you run google internal tests as well to make sure it does not break anything.

**Comment by Cjkkkk** [comment]
@ezhulenev could you share the some details of the internal check failure?

**Pull request closed without merging**